### ISSUE TITLE
[Bug] Resolving Android dependencies: Job failed error

### ISSUE THREAD

**Issue by idamaeng** [firebase/firebase-unity-sdk#1117]
### Description

When trying to resolve dependencies for android, the result is an exception:

`Job failed with exception: System.ArgumentException: The specified path is not of a legal form (empty).
  at System.IO.Path.InsecureGetFullPath (System.String path) [0x00025] in <a6d715f5ce8c4754a140b50ff638878a>:0 
  at System.IO.Path.GetFullPath (System.String path) [0x00000] in <a6d715f5ce8c4754a140b50ff638878a>:0 
  at System.IO.DirectoryInfo..ctor (System.String path) [0x00006] in <a6d715f5ce8c4754a140b50ff638878a>:0 
  at (wrapper remoting-invoke-with-check) System.IO.DirectoryInfo..ctor(string)
  at Google.FileUtils.CreateFolder (System.String path, Google.Logger logger) [0x00010] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:646 
  at Google.FileUtils.CreateFolder (System.String path, Google.Logger logger) [0x0001e] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:648 
  at Google.FileUtils.CreateFolder (System.String path, Google.Logger logger) [0x0001e] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:648 
  at Google.FileUtils.CreateFolder (System.String path, Google.Logger logger) [0x0001e] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:648 
  at Google.FileUtils.CreateFolder (System.String path, Google.Logger logger) [0x0001e] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:648 
  at Google.FileUtils.CreateFolder (System.String path, Google.Logger logger) [0x0001e] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:648 
  at Google.FileUtils.CreateFolder (System.String path, Google.Logger logger) [0x0001e] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:648 
  at Google.FileUtils.CreateFolder (System.String path, Google.Logger logger) [0x0001e] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:648 
  at Google.FileUtils.CreateFolder (System.String path, Google.Logger logger) [0x0001e] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:648 
  at Google.FileUtils.CreateFolder (System.String path, Google.Logger logger) [0x0001e] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:648 
  at Google.FileUtils.CreateFolder (System.String path, Google.Logger logger) [0x0001e] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/VersionHandlerImpl/src/FileUtils.cs:648 
  at GooglePlayServices.PlayServicesResolver.CopyAssetAndLabel (System.String sourceLocation, System.String targetLocation, System.Boolean force) [0x0009b] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:2514 
  at GooglePlayServices.GradleTemplateResolver.CopySrcAars (System.Collections.Generic.ICollection`1[T] dependencies) [0x000be] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/AndroidResolver/src/GradleTemplateResolver.cs:212 
  at GooglePlayServices.GradleTemplateResolver.InjectDependencies (System.Collections.Generic.ICollection`1[T] dependencies) [0x00140] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/AndroidResolver/src/GradleTemplateResolver.cs:551 
  at GooglePlayServices.PlayServicesResolver+<>c__DisplayClass115_0.<ResolveUnsafe>b__3 () [0x0000c] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/AndroidResolver/src/PlayServicesResolver.cs:2078 
  at Google.RunOnMainThread.ExecuteNext () [0x00047] in /Users/amaurice/Documents/GitHub/unity-jar-resolver/source/VersionHandlerImpl/src/RunOnMainThread.cs:486 `

These are the firebase sdks I use

![Screenshot 2024-09-25 at 17 37 56](https://github.com/user-attachments/assets/ef611901-ecb4-4e30-aeb5-072bf50b9139)

Thanks in advance!

### Reproducing the issue

Create an empty unity project with 6000.0.20f1 add sdks, use edm4u to resolve dependencies.

### Firebase Unity SDK Version

12.0.0

### Unity editor version

6000.0.20f1

### Installation Method

Unity Package Manager

### Problematic Firebase Component(s)

All

### Targeted Platform(s)

Apple Platforms, Android

### Unity editor platform

Mac

### Scripting Runtime

IL2CPP

### Release Distribution Type

Open-source from this repo

### If using CocoaPods for Apple platforms, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

👀 Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by AlexanderLukashin** [comment]
This has been fixed in the latest version
https://github.com/googlesamples/unity-jar-resolver/releases/tag/v1.2.183

**Comment by argzdev** [comment]
Hey @idamaeng, thanks for reaching out. @AlexanderLukashin is correct. This is most likely similar to the [issue](https://github.com/googlesamples/unity-jar-resolver/issues/697) filed in our jar resolver. This happens whenever dependency packages lives in a folder parallel to `Assets` folder. That said, this has been fixed in our External Dependency Manager version `1.2.183`. I tried testing this and so far, no issue is present. I'll go ahead and close this thread for now. Feel free to let us know if you encounter this issue again in the latest version. Thanks!
